### PR TITLE
fix!: Replace 'rds' with 'sql' throughout, including public API

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

Replace public instances of **RDS** with **SQL** to avoid confusion about the solution's capabilities. This should encompass all public APIs, as reported by the various `API.md` changes.

#### Description of how you validated changes

Manual test & build. Will run e2e tests to verify.

This change must be deployed in conjunction with the SQL layer changes tracked elsewhere.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by palpatim — COMMENTED** [review]
marker review

File: packages/graphql-transformer-common/src/ResourceConstants.ts
Comment (alharris-at) [inline packages/graphql-transformer-common/src/ResourceConstants.ts:41]: Asking since these cross the boundary into the V1 package namespace. I assume so, but are all of these config values added after we started on gqlv2 rds? Or is it possible these are expected to be inputs to the old v1 relational stuff? https://docs.amplify.aws/cli-legacy/graphql-transformer/relational/

**Review by alharris-at — DISMISSED** [review]
Overall LGTM, just one question on the non-Amplify prefixed packages.

File: packages/amplify-category-api/src/commands/api/add-graphql-datasource.ts
Comment (AaronZyLee) [inline packages/amplify-category-api/src/commands/api/add-graphql-datasource.ts:127]: nit: The line 130 still mentions about `RDS schema` which can be `SQL schema` according to the PR's theme.

**Review by AaronZyLee — APPROVED** [review]

File: packages/amplify-category-api/src/commands/api/add-graphql-datasource.ts
Comment (palpatim) [reply packages/amplify-category-api/src/commands/api/add-graphql-datasource.ts:127]: Thanks for catching. Will follow up in a subsequent PR, so as not to dismiss these reviews & e2e test results (assuming they pass)

File: packages/graphql-transformer-common/src/ResourceConstants.ts
Comment (palpatim) [reply packages/graphql-transformer-common/src/ResourceConstants.ts:41]: I believe I isolated these to only new V2 symbols & values. I explicitly did NOT change the items relating to previously-existing RDS Data Source support.

**Comment by palpatim** [comment]
Rebase/f
orce pushed to pick up latest merges from `main` - https://github.com/aws-amplify/amplify-category-api/pull/2030

File: packages/amplify-graphql-schema-generator/API.md
Comment (dpilch) [inline packages/amplify-graphql-schema-generator/API.md:164]: Do we need to change this to `graphqlSchemaFromSQLSchema`?

**Comment by palpatim** [comment]
Need to redo all of this after the merge, closing in favor of a new one later.

**Pull request closed without merging**
